lib/mayaUsd/ufe/StagesSubject.cpp

```diff
@@ -258,7 +258,13 @@ void StagesSubject::stageChanged(UsdNotice::ObjectsChanged const& notice, UsdSta
 #ifdef UFE_V2_FEATURES_AVAILABLE
 		else if (!prim.IsValid() && !InPathChange::inPathChange())
 		{
-			if (InAddOrDeleteOperation::inAddOrDeleteOperation())
```

Comment:
Hmm, this is quite strange, and is reversing the logic.  The idea was that we generate a SubtreeInvalidate notification unless we know better, i.e. unless we know we're in an add or delete operation (here delete, I would hope).  Therefore, SubtreeInvalidate is the default, and delete will only be sent if the code implementing the delete has put in a delete guard.

Reply:
Thanks for your review @ppt-adsk . I hope I didn't mess up the logic. What's happening is that we are getting a delete notification, but we did not initiate this delete and therefore inAddOrDeleteOperation is false.  The deletion was caused by a reload() on a layer.  sceneItem will be remain null if InaddOrDeleteOperation is true, so it should be passing to the ObjectDestroyed notification as before.  createItem will return nullptr in the case of this deletion notification, and therefore it will also go to the ObjectDestroyed notification.

Reply:
I would just move the createItem above the original line 261 and then change 261 to

`if (!sceneItem || InAddOrDeleteOperation::inAddOrDeleteOperation())`

Reply:
@ppt-adsk The case we are trying to catch here is when the prim was deleted by outside sources and not going thru our delete operations. In this case since the prim is deleted, then createItem() returns nullptr which is then packaged up in the SubtreeInvalidate message. The Outliner gets it, expects a valid item to invalidate and crashes because its null.